src/mame/midway/midyunit_m.cpp

```diff
@@ -628,23 +629,41 @@ void midyunit_state::midyunit_sound_w(offs_t offset, uint16_t data, uint16_t mem
 
 	// call through based on the sound type
 	if (ACCESSING_BITS_0_7 && ACCESSING_BITS_8_15)
-		switch (m_chip_type)
-		{
-			case SOUND_NARC:
-				m_narc_sound->write(data);
-				break;
-
-			case SOUND_CVSD_SMALL:
-			case SOUND_CVSD:
-				m_cvsd_sound->reset_write((~data & 0x100) >> 8);
-				m_cvsd_sound->write((data & 0xff) | ((data & 0x200) >> 1));
-				break;
-
-			case SOUND_ADPCM:
-				m_adpcm_sound->reset_write((~data & 0x100) >> 8);
-				m_adpcm_sound->write(data);
-				break;
-		}
+		m_narc_sound->write(data);
+}
+
+void midyunit_cvsd_state::cvsd_sound_w(offs_t offset, uint16_t data, uint16_t mem_mask)
+{
+	// check for out-of-bounds accesses
+	if (offset)
+	{
+		logerror("%08X:Unexpected write to sound (hi) = %04X\n", m_maincpu->pc(), data);
+		return;
+	}
+
+	// call through based on the sound type
+	if (ACCESSING_BITS_0_7 && ACCESSING_BITS_8_15)
+	{
+		m_cvsd_sound->reset_write((~data & 0x100) >> 8);
```

Comment:
You could use `BIT(~data, 8)` here to avoid needing two constants that have to line up.